lib/overcommit/hook/pre_commit/php_stan.rb

```diff
@@ -0,0 +1,30 @@
+module Overcommit::Hook::PreCommit
+  # Runs `phpstan` against any modified PHP files.
+  # For running `phpstan` with Laravel, it requires setup with `ide_helper`.
+  #
+  # References:
+  # https://github.com/phpstan/phpstan/issues/239
+  # https://gist.github.com/edmondscommerce/89695c9cd2584fefdf540fb1c528d2c2
+  class PhpStan < Base
+    MESSAGE_REGEX = /^(?<file>.+)\:(?<line>\d+)\:(?<message>.+)/
+
+    def run
+      messages = []
+
+      applicable_files.each do |file|
+        result = execute(command, args: [file])
```

Comment:
Is there a reason you're only running one file at a time? This will run very slowly for a large number of files, since you're running them in serial.

Unless there's something about how the tool manages this, would strongly recommend you pass all files to `phpstan` in one run.

Reply:
Will update. It does make sense.